### PR TITLE
Complete PCA9685 Code

### PR DESCRIPTION
**Pull request by Roastlawyer**
I've redone my PCA9685 code so that its now in the same format as Servo.H and ESP32Servo.h

I did have to still insert a setup line in the opengloves.ino file as I could not get initialization to start without one.

### CODE REVIEW

**Comment by JohnRThomas** [comment]
Can you squash into a single commit please, it helps keep the git tree clean and concise and ties the exact change to a commit message.

File: open-gloves/HardwareConfig.hpp [hunk 1]
```diff
@@ -12,6 +12,10 @@
 #include "LED.hpp"
 #include "Pin.hpp"
 
+#if ENABLE_PCA_9865_SERVO
+  #include <Wire.h>
```
Comment (JohnRThomas) [inline open-gloves/HardwareConfig.hpp:16]: Is this a standard Arduino header? If so, you can remove the ifdefs.

File: open-gloves/Config.h [hunk 2]
```diff
@@ -189,12 +191,23 @@
...
 #define MEDIAN_SAMPLES 20
 
-// You must install Seeed-PCA9685 Library
-// https://github.com/Seeed-Studio/Seeed_PCA9685
+
+
+// You must install the Adafruit PCA9685 PWM Servo Driver Library 
+// https://github.com/adafruit/Adafruit-PWM-Servo-Driver-Library
+
 #define ENABLE_PCA_9865_SERVO true
```
Comment (JohnRThomas) [inline open-gloves/Config.h:199]: this should be false by default

File: open-gloves/Config.h [hunk 3]
```diff
@@ -189,12 +191,23 @@
...
+  #define ServoMax_uS                   2400    // Max microsecond pulse length, moves servo to 180* or whatever it's max rotation position is. Value for sg90 and mg90s by default (ESP32Servo default is 2500)
 #endif
-#define PIN_I2C_SDA         16
-#define PIN_I2C_SCL         22
+
+
+ // microsecond value to 
+
+
+#define PIN_I2C_SDA         18
```
Comment (JohnRThomas) [inline open-gloves/Config.h:212]: set these pins to pin 0 by default

File: open-gloves/PCA9685Servo.hpp [hunk 4]
```diff
@@ -2,28 +2,58 @@
...
   }
+  this->writeMicroseconds(value);
+}
 
- private:
-  int channel;
-};
\ No newline at end of file
+void Servo::writeMicroseconds(int value){
+  //TODO: (Roastlawyer) force all inputs to always be mapped between min and max microseconds
```
Comment (JohnRThomas) [inline open-gloves/PCA9685Servo.hpp:57]: Please complete any TODOs before merging.

**Review by JohnRThomas — CHANGES_REQUESTED** [review]

File: open-gloves/Config.h [hunk 5]
```diff
@@ -176,11 +176,13 @@
...
   #define PIN_LED             2
-  #define PIN_PINKY_FFB       5  //used for force feedback
-  #define PIN_RING_FFB        18 //^
-  #define PIN_MIDDLE_FFB      19 //^
-  #define PIN_INDEX_FFB       21 //^
-  #define PIN_THUMB_FFB       17 //^
+  
+  #define PIN_PINKY_FFB       15  //used for force feedback
+  #define PIN_RING_FFB        14 //^
+  #define PIN_MIDDLE_FFB      13 //^
```
[the remark this answers is not in the document]
Comment (Roastlawyer) [reply open-gloves/Config.h]: Done

File: open-gloves/HardwareConfig.hpp (hunk 1 above)
Comment (Roastlawyer) [reply open-gloves/HardwareConfig.hpp:16]: Wire.h is not normally included with the sketch. That line causes it to only be included if the PCA 9685 is going to be used.

File: open-gloves/Config.h (hunk 2 above)
Comment (Roastlawyer) [reply open-gloves/Config.h]: Fixed

File: open-gloves/Config.h (hunk 3 above)
Comment (Roastlawyer) [reply open-gloves/Config.h]: Has been set

File: open-gloves/PCA9685Servo.hpp (hunk 4 above)
Comment (Roastlawyer) [reply open-gloves/PCA9685Servo.hpp]: When I had investigated further I learned that the TODO was redundant. The smooth stepping code already uses MIN_PULSE_WIDTH & MAX_PULSE_WIDTH anyways, it was just not configurable before.  Regular degree based servo control is mapped in the previous few lines, meaning that tall inputs to the writeMicroseconds function where already mapped to the range previously. Removed the TODO.

**Comment by Roastlawyer** [comment]
Second force push was to remove 2 blank lines accidentally added in the middle of Config.h

**Comment by Roastlawyer** [comment]
Additional force-push was to remove some blank lines accidentally added to config.h

**Review by JohnRThomas — APPROVED** [review]

**Pull request merged**